src/Console/Command/CompareCommand.php

```diff
@@ -0,0 +1,173 @@
+use Symfony\Component\Console\Helper\Table;
+use Symfony\Component\Console\Helper\TableCell;
+use Symfony\Component\Console\Input\InputInterface;
+use Symfony\Component\Console\Input\InputOption;
+use Symfony\Component\Console\Output\OutputInterface;
+
+/**
+ * @author Adamo Aerendir Crespi <hello@aerendir.me>
+ */
+final class CompareCommand extends Command
+{
+    const COMMAND_NAME = 'compare';
+
+    /**
+     * @var ConfigInterface
+     */
+    private $defaultConfig;
+
+    /**
+     * @var FixerFactory
+     */
+    private $fixerFactory;
+
+    /**
+     * @param null|FixerFactory $fixerFactory
+     */
+    public function __construct(FixerFactory $fixerFactory = null)
+    {
+        parent::__construct();
+
+        if (null === $fixerFactory) {
+            $fixerFactory = new FixerFactory();
+            $fixerFactory->registerBuiltInFixers();
+        }
+
+        $this->fixerFactory = $fixerFactory;
+        $this->defaultConfig = new Config();
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    protected function configure()
+    {
+        $this
+            ->setName(self::COMMAND_NAME)
+            ->setDefinition(
+                [
+                    new InputOption('config', '', InputOption::VALUE_REQUIRED, 'The path to a .php_cs file.'),
+                    new InputOption('show-risky', '', InputOption::VALUE_NONE, 'Shows also the riky fixers.'),
+                    new InputOption('hide-in-use', '', InputOption::VALUE_NONE, 'Hides all the rules currently configured (and so, used) to highlight only the ones not already in use.'),
+                    new InputOption('dump', '', InputOption::VALUE_NONE, 'Dumps the comparing result in a copy-and-pastable format ready for the .php_cs file.'),
+                ]
+            )
+            ->setDescription('Compares existent features with the ones actually configured.')
+        ;
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    protected function execute(InputInterface $input, OutputInterface $output)
+    {
+        $passedConfig = $input->getOption('config');
+
+        $resolver = new ConfigurationResolver(
+            $this->defaultConfig,
+            [
+                'allow-risky' => true,
+                'config' => $passedConfig,
+            ],
+            getcwd()
+        );
+
+        $inUse = $resolver->getRules();
+        $configured = $resolver->getConfig()->getRules();
+        $builtIn = $this->fixerFactory->getFixers();
+
+        usort($builtIn, function ($a, $b) {
+            return strcmp($a->getName(), $b->getName());
+        });
+
+        $rows = [];
+        $dump = [];
+        $builtInCount = 0;
+        foreach ($builtIn as $fixer) {
+            ++$builtInCount;
+
+            if ($fixer->isRisky() && false === $input->getOption('show-risky')) {
+                // Don't show risky fixers if required
+                continue;
+            }
+
+            $isUsed = array_key_exists($fixer->getName(), $inUse);
```

Comment:
if user has ruleset `[@Symfony => true, other_rules => ...]` and then see that he is using `braces` rule, he has no clue why.
we shall display where the rule "in use" comes from, like it is active as user requested it directly or it's part of set he is using

Reply:
Yes, this is a very good point. I were thinking at it yesterday, but I have no idea where to get this info: can you give me some advice?

PS I'll ignore all other points as I'm completely refactoring the command. Once I'll commit the modifications, we'll can start to improve it...

Reply:
Done.